front/components/data_source/StandardDataSourceView.tsx

```diff
@@ -0,0 +1,546 @@
+import {
+  Button,
+  ContextItem,
+  Dialog,
+  DocumentTextIcon,
+  Page,
+  PencilSquareIcon,
+  PlusIcon,
+  Popup,
+  ServerIcon,
+  Tab,
+} from "@dust-tt/sparkle";
+import type {
+  CoreAPIDocument,
+  DataSourceType,
+  PlanType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import type { PostDataSourceDocumentRequestBody } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import { useRouter } from "next/router";
+import { useContext, useEffect, useMemo, useRef, useState } from "react";
+
+import { DocumentUploadModal } from "@app/components/data_source/DocumentUploadModal";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { handleFileUploadToText } from "@app/lib/client/handle_file_upload";
+import { tableKey } from "@app/lib/client/tables_query";
+import { getDisplayNameForDocument } from "@app/lib/data_sources";
+import { useDocuments, useTables } from "@app/lib/swr";
+import { ClientSideTracking } from "@app/lib/tracking/client";
+import { timeAgoFrom } from "@app/lib/utils";
+
+const tabIds = ["documents", "tables"];
+
+function DatasourceDocumentsTabView({
```

Comment:
just for readability can you split in multiple files for DatasourceTablesTabView / DatasourceDocumentsTabView and  StandardDataSourceView ?